src/wallet/walletdb.cpp

```diff
@@ -892,6 +892,8 @@ ReadKeyValue(CWallet* pwallet, CDataStream& ssKey, CDataStream& ssValue,
                 strErr = "Error in wallet database: recipientmapping UA does not contain recipient";
                 return false;
             }
+
+            pwallet->LoadRecipientMapping(txid, RecipientMapping(ua.value(), recipient));
```

Comment:
This is silently tucked away in an unrelated commit. Move this into a separate commit with an explanation for this change in the commit message.